plugins/action/action.h

```diff
@@ -246,7 +246,28 @@ class Action : public PluginBase {
      */
     float get_max_speed_m_s() const;
 
-    // Non-copyable
+    /**
+     * @brief Set the return to launch minimum return altitude.
+     *
+     * When return to launch is initiated, the vehicle climbs to the return altitude if it is lower
```

Comment:
1. The naming in PX4 for RTL is now "Return Mode". We should perhaps rename the other API
2. The behaviour on return is configurable based on parameters - e.g. it might not land. In particular if you call this on a Fixed Wing vehicle. I tend to not talk about the landing if I don't need to.
3. Would it make sense to embed the setter into the return command function? Probably not, but we should certainly do a see also between this and the return command function.

Perhaps:

> When Return mode is engaged the vehicle will ascend to this altitude (if below) for the return flight. The vehicle will maintain its current altitude if above this altitude.

Reply:
1. Good point, we should think about the naming.

2. I think the SDK needs to assume default params for everything it has not changed. Otherwise we're opening a whole can of worms and can't guarantee at all what happens when you call something. The firmware has way too many parameters that you need to know to be safe.

3. My aim is to have simple functions with as little arguments as possible. Otherwise, the function signature soon would look like this:
```
void Action::return_to_launch_async(action_callback_t callback, float return_altitude_m, bool include_land, bool do_smart_rtl);
```
```